promql/engine.go

```diff
@@ -2770,19 +2774,28 @@ func vectorElemBinop(op parser.ItemType, lhs, rhs float64, hlhs, hrhs *histogram
 			}
 			return 0, res.Compact(0), true, nil
 		}
-		return lhs - rhs, nil, true, nil
+		if hlhs == nil && hrhs == nil {
+			return lhs - rhs, nil, true, nil
+		}
+		return 0, nil, false, annotations.NewMixedFloatsHistogramsInfo(pos)
 	case parser.MUL:
 		if hlhs != nil && hrhs == nil {
 			return 0, hlhs.Copy().Mul(rhs), true, nil
 		}
 		if hlhs == nil && hrhs != nil {
 			return 0, hrhs.Copy().Mul(lhs), true, nil
 		}
+		if hlhs != nil && hrhs != nil {
+			return 0, nil, false, annotations.NewMixedFloatsHistogramsInfo(pos)
```

Comment:
The problem here is actually not the _mix_ of float and histogram, but that we have _only_ histograms. So we need a different annotation for that case.

Reply:
Yeah, I think it would be better to have the annotation something like NewInvalidHistogramsBinOp. WDYT?

Reply:
Yes, we can try that. However, in the end, it would be good to get an error message specific enough so that the user can guess what's going wrong. So maybe have a different one for "this operation is not valid between two histograms" (e.g. for multiplication) and "this operation is not valid between a float and a histogram" (e.g. for addition).